integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -164,14 +165,26 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
           targetCpu = workerCpu;
           targetMem = workerMem;
           mWorkers.add(offer.getHostname());
+          mIsMasterTask = false;
         } else {
           // The resource offer cannot be used to start either master or a worker.
+          driver.declineOffer(offer.getId());
           continue;
         }
+        
+        mTaskName = mIsMasterTask ? sConf.get(Constants.INTEGRATION_MESOS_TACHYON_MASTER_NAME) : 
```

Comment:
if the mTaskName assignments above are implemented, you can get rid of this line